### DISCUSSION TITLE
Citing Neurodesk before we have a preprint out

### DISCUSSION THREAD

**Discussion by civier** [neurodesk/neurodesk.github.io#114] [Q&A] [answered]
@Joanone will submit a paper next week and she wishes to cite Neurodesk.
Since we still don't have a preprint out, I hoped that we can discuss it in our next meeting, but it will be too late.
**Which abstract should she cite in the time being? Or do you have a better idea?**

@Joanone, could you maybe let us know where you are going to submit the manuscript to? and also, can you please confirm that you mainly used the diffusion MRI tools?

**Comment by Joanone** [comment]
Hello everyone,

Yes, i'm planning to submit a paper for a conference proceeding.

The update to this is that, the deadline is in the last week of April, i would probably submit within the first/second week of April. So i think there is more time to know which Neurodesk-related article to cite.

Its an international conference on Neuroinformatics, Yes, the major part of my analysis was performed using the Mrtrix3 software on Neurodesk, i also used afni and fsl within the environment too, other parts of the analysis i performed externally was on Matlab.

**Reply by civier** [reply]
Dear @Joanone 

Very exciting that you've successfully used 3 different tools within NeuroDesk. 

Should we make it 4 ? :-) What I want to suggest is that you give our current MATLAB solution a shot. We have two packages that are compiled MATLAB code (thus, you can use them without a MATLAB licence), and moreover, they allow you to run external scripts. So if you already have MATLAB scripts that you tested, you can simply run them via one of these packages.
The instructions on how to run scripts from the FieldTrip package is below, so would be super if you could give it a try and report how it goes. Notice that you don't need to utilise any FieldTrip functions in your scripts or anything like that -- FieldTrip provides the ability to run any general script, and we just use it as a platform to launch scripts.
https://neurodesk.github.io/tutorials/electrophysiology/fieldtrip/
The premise is that you would be able to run your analysis fully in NeuroDesk, which would be even more impressive. It will also enable others to reproduce your pipeline in NeuroDesk without them needing to have a MATLAB licence! 

I hope that it's not too much to ask, and let me know how it goes (if the MATLAB version used by the FieldTrip container is too old, let us know what version your MATLAB scripts require, and we'll see what can be done)

All the best,
Oren

**Reply by Joanone** [reply]
Okay Oren, i can give this a shot in the coming week.

How can i check the MATLAB version FieldTrip container uses? I currently use the 2020 version and i would like to use the same in testing it.

**Reply by stebo85** [reply]
Dear @civier and @Joanone,

I would not recommend to try to run arbitrary scripts with the mcr as there are a couple of differences and some problems can be difficult to find (caused by fixed path orders and only specific toolboxes will be included that were selected during compile time):
https://au.mathworks.com/help/compiler/differences-between-matlab-and-matlab-runtime.html

**Reply by stebo85** [reply]
@Joanone - you can find which MCR version was used for the FieldTrip container here: https://github.com/NeuroDesk/neurocontainers/blob/7cddfae13903ffe737dcd0946334c401818e96f6/recipes/fieldtrip/build.sh#L28 - it's 2020b

**Comment by civier** [comment]
Thanks so much @Joanone for giving it a try.
I guess you can just follow the tutorial, and provide a one-line script that prints the MATLAB version using the following command:
https://www.mathworks.com/help/matlab/ref/ver.html

Let me know if it doesn't work, and I can check how the container was built (we keep the install scripts for all software available in Neurodesk exactly for such situations). In reality, it would be best if such information were available in https://neurodesk.github.io/applications/ next to each package name, but we haven't implemented it yet. If you would like to contribute to Neurodesk even more (always appreciated), that can be a possible task for you to work on.

(@JD-Zhu If you remember what version of Matlab you used for the FieldTrip container, you're welcome to pitch in)

**Comment by JD-Zhu** [comment]
I think we used Matlab R2020b for the FieldTrip container :)

**Comment by stebo85** [comment, marked as answer]
Dear @Joanone - to answer your original question here:
Please just cite our website for now: http://neurodesk.github.io/ 

More important than citing NeuroDesk is actually to cite the tools you used! So if you used Mrtrix3 make sure you cite the papers that the authors of Mrtrix3 ask you cite and the respective methods you used! Same for AFNI and FSL! I started adding citation recommendations to the READMEs of the containers:

Mrtrix3:
J.-D. Tournier, R. E. Smith, D. Raffelt, R. Tabbara, T. Dhollander, M. Pietsch, D. Christiaens, B. Jeurissen, C.-H. Yeh, and A. Connelly. MRtrix3: A fast, flexible and open software framework for medical image processing and visualisation. NeuroImage, 202 (2019), pp. 116–37.


FSL: 

1. M.W. Woolrich, S. Jbabdi, B. Patenaude, M. Chappell, S. Makni, T. Behrens, C. Beckmann, M. Jenkinson, S.M. Smith. Bayesian analysis of neuroimaging data in FSL. NeuroImage, 45:S173-86, 2009

2. S.M. Smith, M. Jenkinson, M.W. Woolrich, C.F. Beckmann, T.E.J. Behrens, H. Johansen-Berg, P.R. Bannister, M. De Luca, I. Drobnjak, D.E. Flitney, R. Niazy, J. Saunders, J. Vickers, Y. Zhang, N. De Stefano, J.M. Brady, and P.M. Matthews. Advances in functional and structural MR image analysis and implementation as FSL. NeuroImage, 23(S1):208-19, 2004

3. M. Jenkinson, C.F. Beckmann, T.E. Behrens, M.W. Woolrich, S.M. Smith. FSL. NeuroImage, 62:782-90, 2012

AFNI:
Cox RW (1996). AFNI: software for analysis and visualization of functional magnetic resonance neuroimages. Comput Biomed Res 29(3):162-173. doi:10.1006/cbmr.1996.0014
https://pubmed.ncbi.nlm.nih.gov/8812068/

RW Cox, JS Hyde (1997). Software tools for analysis and visualization of FMRI Data. NMR in Biomedicine, 10: 171-178.
https://pubmed.ncbi.nlm.nih.gov/9430344/

**Reply by Joanone** [reply]
Okay @stebo85 . Thanks!
